### PR TITLE
Allow Importing workspaces using workspace name

### PR DESCRIPTION
**Pull request by sebasslash**
## Description

This PR adds support for importing workspaces using a `<ORGANIZATION NAME>/<WORKSPACE NAME>` nested combo. You can still pass a workspace external id.

An example import command looks as follows:

```sh
terraform import tfe_workspace.test my-org-name/my-wkspace-name
```

### How to test
**To run acceptance tests**:

```sh
TESTARGS="-run TestAccTFEWorkspace_import" make testacc
```

**To smoke test**:

1. Create a Terraform configuration file, setting the tfe provider and the workspace resource we'll import to.

```hcl
provider "tfe" {
  hostname = "my-tfe-host"
  token = "my-tfe-token
}

resource "tfe_workspace" "test" {}
```

Ensure you have the dev overrides set to the compiled binary of this PR in your `go/bin`. 

2. Create an organization in your TFE host and workspace.
3. Import!

```sh
terraform import tfe_workspace.test org-name/wkspace-name
```

### CODE REVIEW

File: tfe/resource_tfe_workspace.go [hunk 1]
```diff
@@ -642,3 +642,26 @@ func validateRemoteState(_ context.Context, d *schema.ResourceDiff, meta interfa
 
 	return nil
 }
+
+func resourceTFEWorkspaceImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	tfeClient := meta.(*tfe.Client)
+
+	s := strings.SplitN(d.Id(), "/", 2)
+	if len(s) >= 3 {
```
Comment (brandonc) [inline tfe/resource_tfe_workspace.go:650]: At most 2 strings will be returned by `strings.SplitN(d.Id(), "/", 2)`

File: tfe/resource_tfe_workspace.go [hunk 2]
```diff
@@ -642,3 +642,26 @@ func validateRemoteState(_ context.Context, d *schema.ResourceDiff, meta interfa
...
+		)
+	} else if len(s) == 2 {
+		workspaceID, err := fetchWorkspaceExternalID(s[0]+"/"+s[1], tfeClient)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"error retrieving workspace with name %s from organization %s %v", s[1], s[0], err)
+		}
+
+		d.SetId(workspaceID)
+		return []*schema.ResourceData{d}, nil
```
Comment (brandonc) [inline tfe/resource_tfe_workspace.go:663]: This line can be omitted

**Review by brandonc — COMMENTED** [review]
I tried this and it worked with both methods. I think you can test import by using the ImportState/ImportStateId property of resource.TestStep.

**Review by brandonc — APPROVED** [review]

**Comment by chrisarcand** [comment]
Nice! 👏🏻

**Pull request merged**